### ISSUE TITLE
Issue with selection of 15khz mode for CRT use

### ISSUE THREAD

**Issue by z00p** [libretro/px68k-libretro#70]
I'm using the RetroArch CRT EmuDriver options on a Sony multi-sync monitor. Getting px68k to display 15khz (240p) mode is hit and miss. Currently (with Dx10 or Dx11 driver + CRT enabled) X68k will start in 720p mode for the X68000 boot and in game. To get into 240p mode I need to switch into the px68k menu using F12, then when I go back from the menu into the game it switches to correct resolution (240p) and looks amazing.

**Comment by z00p** [comment]
Right. The process of going in and out of the PK68k menu switches the video from from 720p (set when the emulator starts) to 240p (480i).

**Comment by gingerbeardman** [comment]
Something must not be getting initialised correctly. I'll try to take a look this week.

**Comment by retro-wertz** [comment]
menu is 800x600, dos prompt mostly triggers @ 768x512. some games are 256x256 or x2 of that with others at 768x512.

im revamping the core (for personal use, i hate this ever changing screensize and a compromise of fixed 800x600 window geometry works for me best-letterbox or borders does not bother me if its main screen is big enough, frames are still rendered natively)

how does the crt-driver work with height larger than 256? and does it need to be equal or less than that.

**Comment by negativeExponent** [comment]
i hope i had a CRT and a way to connect to it. anyways does using core overrides and running core at a fixed frame helps in any way?

**Comment by riggles1** [comment]
@negativeExponent I'm not entirely sure. It does perfectly select the 2560 superres 256p55Hz mode with CRTSwitchres (needs a CRT capable of 55Hz vertical sync and geometry adjustments to display all 256 lines of graphics properly).

But selecting other modes within games that support them doesn't seem to work, for example the 240p mode (60Hz) in Cotton, or the 15-25-31kHz selections in Keeper, which is an oddball game that displays 240p60 instead of 256p55.

I just ended up turning off auto CRTSwitchres (INI>OFF) and forcing the required mode manually, saving the config for the specific game.

But ultimately someone that has all 15-25-31kHz modelines should chime in. I'm just able to do 256p 55Hz and other 15kHz total scanning frequency modes (caps out at 16.2kHz or so).

**Comment by Tatsuya79** [comment]
I'm noticing a sync issue I'm having.
I use gsync with "sync to exact content framerate" but when I'm starting a game, like chourensha here, I'm at 61.45Hz already while booting, and still at that speed in game.

I have the "adjust frame rates" core option disabled as I want the original machine refresh.
If I enable it and then disable it, it stays at 55.45Hz then.

Did something change in the core regarding that? I tried a 2 years old version of retroarch and it acts the same.
I think it was working before.

**Comment by negativeExponent** [comment]
yeah, this is indeed **int VID_MODE = MODE_HIGH; /* what framerate we start in */** in my port. if it works fine this way in your end, then lets just make this as default then.

**Comment by Tatsuya79** [comment]
Great, that fixes it. 👍 

I'm still noticing a case where the timing is wrong: when you load a save state made at a different refresh than the default.

Like you go screen mode 2 in chourensha for 61.45hz, do a save state: when you'll restart the game later and load it you'll end up at 55.45hz with a slowed down game.

**Comment by negativeExponent** [comment]
can you add SFVAR(VID_MODE), to crtc StateRegs and see if this helps?

**Comment by Tatsuya79** [comment]
This is working. 👍 
Run ahead seems to be performing the same comparing before/after.
